crates/editor/src/editor.rs

```diff
@@ -9966,13 +9966,38 @@ impl Editor {
             None => return None,
         };
 
-        Some(self.perform_format(project, FormatTrigger::Manual, cx))
+        Some(self.perform_format(project, FormatTrigger::Manual, FormatType::Buffer, cx))
+    }
+
+    fn format_selection(
+        &mut self,
+        _: &FormatSelection,
+        cx: &mut ViewContext<Self>,
+    ) -> Option<Task<Result<()>>> {
+        let project = match &self.project {
+            Some(project) => project.clone(),
+            None => return None,
+        };
+
+        let selections = self
```

Comment:
Is there an ideomatic way to get a multiselection without a cursor with a single char?